### PR TITLE
Multiple env and delete unused targets

### PR DESCRIPTION
**Pull request by felire**
## Summary

Multiple Env for Dev, Qa, Stage and Production for ios and android. And the script will also create hte .env files in the project folder. I've also created an script to create the differents build configurations, delete unused targets and I moved the crashlytics config and googleServices config into the script to avoid have unnecessary stuff in the Podfile

## IOS: 

```
Dev - Scheme: dev
Dev - Configuration: Debug
QA - Scheme: qa
QA - Configurations: Debug or QA
Stage - Scheme: stage
Stage - Configurations: Debug or Stage
Production - Scheme: production
Production - Configurations: Debug or Production
```
Examples

```
react-native run-ios --scheme="dev"
react-native run-ios --scheme="qa"
react-native run-ios --scheme="stage"
react-native run-ios --scheme="production"
```

## Android:

```
Dev/QA - Debug: qaDebug
Dev/QA - Release: qaRelease
Stage - Debug: stageDebug
Stage - Release: stageRelease
Production - Debug: productionDebug
Production - Release: productionRelease
```

Example of using:

```
react-native run-android --variant=qaDebug
```

## Trello Card
[https://trello.com/c/B8iwxkJm/19-different-environments-for-dev-qa-stage-and-production](https://trello.com/c/B8iwxkJm/19-different-environments-for-dev-qa-stage-and-production)
[https://trello.com/c/iKBaC8vt](https://trello.com/c/iKBaC8vt)

### CODE REVIEW

File: generators/app/tasks/appSetup/multipleEnvIos/createSchemes.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+const schemeBase = require('./schemeBase');
+
+module.exports = function createSchemes() {
+  const commonPath = schemeName =>
+    `${this.projectName}/ios/${this.projectName}.xcodeproj/xcshareddata/xcschemes/${schemeName}.xcscheme`;
+  this.fs.write(commonPath('qa'), schemeBase.bind(this)('dev'));
```
Comment (ignaciosantise) [inline generators/app/tasks/appSetup/multipleEnvIos/createSchemes.js:6]: is this line ok? or should it be `schemeBase.bind(this)('qa')`?

Comment (felire) [reply generators/app/tasks/appSetup/multipleEnvIos/createSchemes.js:6]: They are different schemes but with the same .env file

Comment (felire) [reply generators/app/tasks/appSetup/multipleEnvIos/createSchemes.js:6]: We'd have "qa" scheme and "dev" scheme and both would use the `.env.dev` file

Comment (guidoprinc) [inline generators/app/tasks/appSetup/multipleEnvIos/createSchemes.js:6]: is this ok?

Comment (felire) [reply generators/app/tasks/appSetup/multipleEnvIos/createSchemes.js:6]: We'd have "qa" scheme and "dev" scheme and both would use the `.env.dev` file

File: generators/app/tasks/appSetup/androidProjectSetup.js [hunk 2]
```diff
@@ -2,7 +2,7 @@ function addDotenvPluginAndRNScreen() {
   const buildGradleContent = this.fs.read(`${this.projectName}/android/app/build.gradle`);
   let updatedBuildGradleContent = buildGradleContent.replace(
     'apply plugin: "com.android.application"',
-    'apply plugin: "com.android.application"\napply from: project(\':react-native-config\').projectDir.getPath() + "/dotenv.gradle"'
+    'apply plugin: "com.android.application"\nproject.ext.envConfigFiles = [qadebug: ".env.dev",\nqarelease: ".env.dev",\nstagedebug: ".env.stage",\nstagerelease: ".env.stage",\nproductiondebug: ".env.production",\nproductionrelease: ".env.production",\nanothercustombuild: ".env"]\napply from: project(\':react-native-config\').projectDir.getPath() + "/dotenv.gradle"'
```
Comment (sfernandez11) [inline generators/app/tasks/appSetup/androidProjectSetup.js:5]: Fix indentation and missing `;` at the end of the Array. Also, remove `anothercustombuild`.

Captura de Pantalla 2020-04-02 a la(s) 12 58 28

File: generators/app/tasks/appSetup/androidProjectSetup.js [hunk 3]
```diff
@@ -16,6 +16,18 @@ function addDotenvPluginAndRNScreen() {
     'if (enableHermes) {',
     "implementation 'androidx.appcompat:appcompat:1.2.0-alpha03'\n\timplementation 'androidx.multidex:multidex:2.0.1'\n\tif (enableHermes) {"
   );
+  updatedBuildGradleContent = updatedBuildGradleContent.replace(
+    'compileSdkVersion rootProject.ext.compileSdkVersion',
+    'compileSdkVersion rootProject.ext.compileSdkVersion\nflavorDimensions "buildtype"'
+  );
```
Comment (sfernandez11) [inline generators/app/tasks/appSetup/androidProjectSetup.js:22]: Same
Captura de Pantalla 2020-04-02 a la(s) 12 58 42

File: generators/app/tasks/appSetup/androidProjectSetup.js [hunk 4]
```diff
@@ -16,6 +16,18 @@ function addDotenvPluginAndRNScreen() {
...
+  updatedBuildGradleContent = updatedBuildGradleContent.replace(
+    '// applicationVariants are e.g. debug, release',
+    'productFlavors {\nqa {\ndimension \'buildtype\'\napplicationIdSuffix ".qa"\n}\nstage {\ndimension \'buildtype\'\napplicationIdSuffix ".stage"\n}\nproduction {\ndimension \'buildtype\'\napplicationIdSuffix ".production"\n}\n}\n// applicationVariants are e.g. debug, release'
+  );
```
Comment (sfernandez11) [inline generators/app/tasks/appSetup/androidProjectSetup.js:26]: Same. Also, the production env doesn't have the `.production` in the end, remove it
Captura de Pantalla 2020-04-02 a la(s) 12 58 59

File: generators/app/tasks/appSetup/androidProjectSetup.js [hunk 5]
```diff
@@ -16,6 +16,18 @@ function addDotenvPluginAndRNScreen() {
...
+  updatedBuildGradleContent = updatedBuildGradleContent.replace(
+    'versionName "1.0"',
+    `versionName "1.0"\nresValue "string", "build_config_package", "com.${this.projectName.toLowerCase()}"`
+  );
```
Comment (sfernandez11) [inline generators/app/tasks/appSetup/androidProjectSetup.js:30]: Same. Also, you forgot to add the `\t` in all cases
Captura de Pantalla 2020-04-02 a la(s) 12 58 47

File: generators/app/tasks/appSetup/createDotEnvFilesLocally.js [hunk 6]
```diff
@@ -0,0 +1,5 @@
+module.exports = function createDotEnvFilesLocally() {
+  this.fs.write(`${this.projectName}/.env.dev`, 'EMPTY_VARIABLE=DEVELOPMENT');
+  this.fs.write(`${this.projectName}/.env.stage`, 'EMPTY_VARIABLE=STAGE');
+  this.fs.write(`${this.projectName}/.env.production`, 'EMPTY_VARIABLE=PRODUCTION');
+};
```
Comment (sfernandez11) [inline generators/app/tasks/appSetup/createDotEnvFilesLocally.js:5]: Add all the `.env`s to `.gitignore` please
```
# Envs 
/.env.*
```

File: CHANGELOG.md [hunk 7]
```diff
@@ -2,6 +2,12 @@
 
 All notable changes to this project will be documented in this file.
 
+## [0.8.0] - 2020-04-XX
+
+- Added multiple env configuration (Debug, QA, Stage and Production) for Android and iOS projects
+- Removed unused targets from iOS project and Podfile (ruby script)
+- Removed temporary fix for the `update_bundle_identifier` Fastlane lane
```
Comment (ignaciosantise) [inline CHANGELOG.md:9]: this is solved :D

Comment (sfernandez11) [reply CHANGELOG.md:9]: I'm going to fix this in the next PR (google-services script)

**Pull request merged**